fhir/src/test/java/org/acme/fhir/MyCamelApplicationTest.java

```diff
@@ -0,0 +1,51 @@
+package org.acme.fhir;
+
+import java.io.File;
+import java.util.concurrent.TimeUnit;
+
+import io.quarkus.test.junit.QuarkusTest;
+import jakarta.inject.Inject;
+import org.apache.camel.CamelContext;
+import org.apache.camel.ProducerTemplate;
+import org.apache.camel.component.mock.MockEndpoint;
+import org.apache.camel.test.infra.fhir.services.FhirService;
+import org.apache.camel.test.infra.fhir.services.FhirServiceFactory;
+import org.apache.commons.io.FileUtils;
+import org.hl7.fhir.r4.model.Patient;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+@QuarkusTest
+public class MyCamelApplicationTest {
+
+    @RegisterExtension
+    public static FhirService service = FhirServiceFactory.createService();
+
+    @Inject
+    CamelContext camelContext;
+
+    @Inject
+    ProducerTemplate producerTemplate;
+
+    @Test
+    public void shouldPushConvertedHl7toFhir() throws Exception {
+        MockEndpoint mock = camelContext.getEndpoint("mock:result", MockEndpoint.class);
+        mock.expectedMessageCount(1);
+
+        // Copy HL7 test data into the test input directory
+        FileUtils.copyDirectory(new File("src/main/data"), new File("target/work/fhir/testinput"));
+
+        // Wait for the file to be processed
+        Thread.sleep(TimeUnit.SECONDS.toMillis(5));
+
+        // Trigger the route manually
+        producerTemplate.sendBody("direct:start", null);
+
+        // Check if the result is as expected
+        mock.assertIsSatisfied();
```

Comment:
This style of testing will not work in native mode. You can create a JAX-RS resource class where you inject the `CamelContext`, then in a service method, use it to get a handle on the `MockEndpoint`.

Then you can reduce the test to just copying the test data file and using `Awaitility.await().untilAsserted()` to poll the JAX-RS endpoint until `MockEndpoint.assertIsSatisfied` runs without error.